### PR TITLE
Fix - Call to undefined function Composer\Autoload\includeFile()

### PR DESCRIPTION
**Pull request by johanjanssens**
Joomla 3.4 is extending [Composer\Autoload\ClassLoader](https://github.com/joomla/joomla-cms/blob/staging/libraries/ClassLoader.php) to make sure JLoader::applyAliasFor($class); is called when a class is being loaded.
### Description

The implementation in Joomla 3.4 prevents to use composer in your own extension and requires developers to [implement a workaround](https://groups.google.com/forum/#!msg/joomla-dev-general/hYZz5JI8ZFA/kK36IV_eRQ0J) first. 

To resolve this the PR rewrites the implementation and instead of extending Composer\Autoload\ClassLoader it decorates it. This is the advised approach to handle this scenario by the Composer team. For details [see following discussion](https://github.com/composer/composer/issues/1493#issuecomment-12492276).

Additionally the fix brings the Composer integration in Joomla more inline with Composer itself. The specialised composer_autoload.php file is removed in favour of the [autoload.php](https://github.com/joomla/joomla-cms/blob/staging/libraries/vendor/autoload.php) file generated by Composer.
### List of changes
- Remove /libraries/composer_autoload.php use default /libraries/vendor/autoload.php as per composer specification
- Decorate \Composer\Autoload\ClassLoader through JClassLoader
- Unregister default class loader and register decorator during Joomla Cms bootstrapping
### Testing instructions

Make sure Joomla still works.

### CODE REVIEW

**Comment by wilsonge** [comment]
Hey Johan. Thanks for this. Looks like a great improvement. A few small things. Firstly travis is failing pretty hard with this change. Secondly can you fix the spaces instead of tabs in the files please :)

**Comment by johanjanssens** [comment]
@wilsonge Changed the files to using tabs instead of spaces as requested. All yours.

File: libraries/classloader.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	 private $loader;
+
+	public function __construct(\Composer\Autoload\ClassLoader $loader)
+	{
+		$this->loader = $loader;
+	}
+	
+	public function loadClass($class)
+	{
+		if($result = $this->loader->loadClass($class)) {
```
Comment (zero-24) [inline libraries/classloader.php]: @johanjanssens can you also fix the if here to:

```
if ($result = $this->loader->loadClass($class))
{
	\JLoader::applyAliasFor($class);
}
```

File: libraries/classloader.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * @since   3.4
+ */
+class JClassLoader
+{
+	/**
+	 * The composer class loader
+	 *
+	 * @var \Composer\Autoload\ClassLoader
+	 */
+	 private $loader;
```
Comment (zero-24) [inline libraries/classloader.php]: @johanjanssens can you also remove the one extra space bevor the `private` :smile:

Comment (johanjanssens) [inline libraries/classloader.php]: @zero-24 Done.

File: libraries/classloader.php (hunk 1 above)
Comment (johanjanssens) [inline libraries/classloader.php]: @zero-24  Done.

**Comment by infograf768** [comment]
To be merged, the deleted files should be added into:
https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_admin/script.php#L339

**Comment by johanjanssens** [comment]
@infograf768 Thanks for the headsup. Made it so. 

Little suggestion to optimise this process. Deleted files and folder could be added to a JSON file during package build. This can be done by getting the info from Git or by comparing packages. A JSON files could be created by the packager to hold the deleted files and folders.

**Comment by zero-24** [comment]
@test just applyed the patch and i can't see that anything breaks i guess joomla will be absolutly broken if this file don't work correct? Or can/need we test some special case?

**Comment by wilsonge** [comment]
We moved everything across to JRegistry so things would still work. You should try using JRegistry instead of /Joomla/Registry/Registry in places and check it still works.

**Comment by zero-24** [comment]
@test i have just tryed some external components (e.g. akeeba backup and ARS es well as others) and they work with that patch.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6335">issues.joomla.org/joomla-cms/6335</a>.</sub>

**Comment by wilsonge** [comment]
Merged in https://github.com/joomla/joomla-cms/commit/e1cfdc0bf85a9905adb7a7627f76af42e788cd44 - thankyou very much @johanjanssens :)

**Comment by johanjanssens** [comment]
@wilsonge Thanks! Great news.

**Comment by johanjanssens** [comment]
resolve joomlatools/joomla-fork#2

**Comment by infograf768** [comment]
please see
http://issues.joomla.org/tracker/joomla-cms/7296

**Pull request closed without merging**